discovery/gossiper.go

```diff
@@ -570,8 +570,26 @@ func New(cfg Config, selfKeyDesc *keychain.KeyDescriptor) *AuthenticatedGossiper
 		futureMsgs:        newFutureMsgCache(maxFutureMessages),
 		quit:              make(chan struct{}),
 		chanPolicyUpdates: make(chan *chanPolicyUpdateRequest),
-		prematureChannelUpdates: lru.NewCache[uint64, *cachedNetworkMsg]( //nolint: ll
+		prematureChannelUpdates: lru.NewCache(
 			maxPrematureUpdates,
+			lru.WithDeleteCallback(
+				func(k uint64, cmsg *cachedNetworkMsg) {
+					// for every network message which is
+					// not processed, send an error to the
+					// channel.
+					cb := func(msg *networkMsg) {
```

Comment:
This function can be a free function, not a closure. It doesn't use any external variables. This would decrease indentation.

Reply:
used it because of the 80 char limit we have. The error becomes almost unreadable.